noxfile.py

```diff
@@ -0,0 +1,34 @@
+import nox
+from nox_poetry import session
+
+
+PYTHON_VERSIONS = ["3.6", "3.7", "3.8", "3.9"]
+SPHINX_VERSIONS = ["2.2", "2.3", "2.4", "3.0", "3.2"]
```

Comment:
@danwos this matrix is, i think, excessive. Do you think this can be stripped down at all?

for reference, github actions allow you to run a matrix in parallel with a maximum of 256 jobs, so we're still nowhere near that limit

Reply:
> But if the test costs nothing and are executed in parallel, lets tests as much as possible.

fair enough, i'll leave it as is for now